mlflow/models/model.py

```diff
@@ -550,8 +559,15 @@ def log(
             local_path = tmp.path("model")
             run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id, metadata=metadata)
-            flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
+            flavor.save_model(
+                path=local_path, mlflow_model=mlflow_model, signature=signature, **kwargs
+            )
             mlflow.tracking.fluent.log_artifacts(local_path, mlflow_model.artifact_path)
+            tracking_uri = _resolve_tracking_uri()
+            if (
+                tracking_uri == "databricks" or get_uri_scheme(tracking_uri) == "databricks"
+            ) and signature is None:
+                _logger.warning(_LOG_MODEL_MISSING_SIGNATURE_WARNING)
             try:
                 mlflow.tracking.fluent._record_logged_model(mlflow_model)
```

Comment:
Note: I decided against adding this warning inside `record_logged_model` because the `mlflow_model` object constructed here does not have access to the model signature.